src/modules/temperature.cpp

```diff
@@ -65,6 +77,17 @@ auto waybar::modules::Temperature::update() -> void {
 }
 
 float waybar::modules::Temperature::getTemperature() {
+#if defined(__FreeBSD__)
+  int temp;
+  size_t size = sizeof temp;
+
+  if (sysctlbyname("hw.acpi.thermal.tz0.temperature", &temp, &size, NULL, 0) != 0) {
+    throw std::runtime_error("sysctl hw.acpi.thermal.tz0.temperature or dev.cpu.0.temperature failed");
```

Comment:
`dev.cpu.0.temperature` doesn't seem to be used here.

Reply:
Hmm, Linux conditional supports more than one `thermal-zone` but here you're hardcoding the first (i.e., `tz0`). No clue if there's a better way to count `tz%d` than to loop `sysctl` until it fails.

Reply:
Ah, you don't need to count. `thermal-zone` number is provided by the user via Waybar config, see man/waybar-temperature.5.scd

Reply:
Thanks,
Done in https://github.com/Alexays/Waybar/pull/1705